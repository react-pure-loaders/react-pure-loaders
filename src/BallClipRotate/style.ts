import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';

const rotate = keyframes({
  '0%': {
    transform: 'rotate(0deg)',
  },
  '100%': {
    transform: 'rotate(360deg)',
  },
  '50%': {
    transform: 'rotate(180deg)',
  },
});

export default (color?: string) => ({
  ...balls(color),
  ...globalAnimation(),
  animation: `${rotate} 0.75s 0s linear infinite`,
  background: 'transparent !important',
  border: `2px solid ${color || PRIMARY_COLOR}`,
  borderBottomColor: 'transparent',
  display: 'inline-block',
  height: '26px',
  width: '26px',
});
