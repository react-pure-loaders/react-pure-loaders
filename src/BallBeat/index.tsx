/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
