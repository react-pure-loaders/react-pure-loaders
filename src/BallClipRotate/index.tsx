/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
