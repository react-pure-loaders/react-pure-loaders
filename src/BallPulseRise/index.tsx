/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
