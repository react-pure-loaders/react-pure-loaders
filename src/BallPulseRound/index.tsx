/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
