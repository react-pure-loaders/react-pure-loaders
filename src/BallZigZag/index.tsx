/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
