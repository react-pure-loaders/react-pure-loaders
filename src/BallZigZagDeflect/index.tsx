/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
