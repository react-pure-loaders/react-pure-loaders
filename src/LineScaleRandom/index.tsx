/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
