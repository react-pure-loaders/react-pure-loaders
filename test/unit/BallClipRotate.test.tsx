import React from 'react';
import Chance from 'chance';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import BallClipRotate from '../../src/BallClipRotate';
import { PRIMARY_COLOR } from '../../src/variables';

const chance = new Chance();

describe('<BallClipRotate>', () => {
    afterEach(cleanup);

    test('BallClipRotate should match snapshot', () => {
        const { container } = render(<BallClipRotate loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallClipRotate should have default color', () => {
        const { getByTestId } = render(<BallClipRotate loading={true}  />);
        const loader = getByTestId('ball-clip-rotate-container').querySelector("div")

        expect(loader)
        .toHaveStyle({
            backgroundColor: PRIMARY_COLOR,
          });
    });

    test('BallClipRotate should have given color', () => {
        const expectedColor = chance.color({ format: 'hex' });
        const { getByTestId } = render(<BallClipRotate loading={true} color={expectedColor}/>);
        const loader = getByTestId('ball-clip-rotate-container').querySelector("div")

        expect(loader)
        .toHaveStyle({
            backgroundColor: expectedColor,
          });
    });

    test('BallClipRotate should have no children', () => {
        const { container } = render(<BallClipRotate loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallClipRotate should have three children', () => {
        const { container } = render(<BallClipRotate loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(2);
    });
});
