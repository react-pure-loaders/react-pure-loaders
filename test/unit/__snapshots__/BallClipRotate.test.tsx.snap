// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallClipRotate> BallClipRotate should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }

  50% {
    -webkit-transform: rotate(180deg);
    -moz-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }
}

.emotion-0 {
  border-radius: 100%;
  height: 26px;
  margin: 2px;
  width: 26px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 0.75s 0s linear infinite;
  animation: animation-0 0.75s 0s linear infinite;
  background: transparent!important;
  border: 2px solid #ffffff;
  border-bottom-color: transparent;
  display: inline-block;
}

<div
  class="emotion-0"
>
  <div />
</div>
`;
